docs/source/machine_translation.rst

```diff
@@ -0,0 +1,255 @@
+`testset <http://ufallab.ms.mff.cuni.cz/~popel/batch3.zip>`_.
+
+Note: Since we are using only small data for example which are not enough
+ for real-life machine translation training.
+
+When we open the archive we find several files for different languages, we
+will use only following files as our training, validation and test set::
+
+    1. ``Batch1a_cs.txt and Batch1a_en.txt`` as our Training set
+    2. ``Batch2a_cs.txt and Batch2a_en.txt`` as a Validation set
+    3. ``Batch3`` as a Test set
+
+Now - before we start, let's make our experiment directory, in which we will
+place all our work. We will call it ``exp-nm-mt``.
+
+First extract all the downloaded files, then make gzip files from individual
+files and put them into following directory structure::
+
+  exp-nm-mt
+  |
+  \== data
+      |
+      |== train
+      |   |
+      |   |== Batch1a_en.txt.gz
+      |   \== Batch1a_cs.txt.gz
+      |
+      |== dev
+      |   |
+      |   |== Batch2a_en.txt.gz
+      |   \== Batch2a_cs.txt.gz
+      |
+      \== test
+          |
+          \== batch3.gz
+
+The gzipping is not a necessary, if you put the dataset there in plaintext it
+ will work the same way. Neural Monkey always try to recognize gzip by the
+extension and opens it.
+
+TODO The dataset is not tokenized and need to be preprocessed.
+
+Byte Pair Encoding
+******************
+
+Neural machine translation (NMT) models typically operate with a fixed
+vocabulary, but translation is an open-vocabulary problem.
+Byte pair encoding (BPE) enables NMT model translation on open-vocabulary by
+encoding rare and unknown words as sequences of subword units.
+This is based on the intuition that various word classes are translatable via
+smaller units than words. More information in the paper
+https://arxiv.org/abs/1508.07909 BPE creates a list of merges that are used
+for splitting out-of-vocabulary words. Example of such splitting::
+
+  basketball => basket@@ ball
+
+Postprocessing can be manually done by::
+
+  sed "s/@@ //g"
+
+BPE generation
+**************
+
+In order to use BPE, you must first generate merge_file, over all data. This
+file is generated on both source and target dataset.
+You can generate it by running following script::
+
+  neuralmonkey/lib/subword_nmt/learn_bpe.py -s 50000 < DATA > merge_file.bpe
+
+With the data from this tutorial it would be the following command::
+
+  paste Batch1a_en.txt Batch1a_cs.txt |
+  neuralmonkey/lib/subword_nmt/learn_bpe.py -s 8000 >
+  exp-nm-mt/data/merge_file.bpe
+
+You can change number of merges, this number is equivalent to the size of the
+vocabulary. Do not forget that as an input is the file containing both source
+and target sides.
+
+
+
+
+Part II. - The Model Configuration
+----------------------------------
+
+In this section, we will go through setting the configuration file
+``translation.ini`` needed for
+the training of the machine translation task. We will mention only
```

Comment:
buď training nebo task, netrénuješ ten task ale ten model
e.g.
needed for the machine translation training